server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/documents.ts

```diff
@@ -44,6 +44,9 @@ export function create(
                 getParam(request.params, "id"));
             documentP.then(
                 (document) => {
+                    if (!document || document.deletionTime) {
```

Comment:
Yes, DeletionTime can be a future timestamp. The property indicates that the document is marked for deletion and specifically when all of its data will be purged from the system. This check is to prevent accessing documents that have been marked for deletion.

Reply:
I'm not sure I understand why we want to prevent access to a document marked for deletion if the deletion will only happen in the future. If the document will only be deleted in, say, 5 days, why should access be prevented until then?

Reply:
This enables us to put the doc in one of 3 states:
1. document functionally accessible - no DeletionTime prop
2. document functionally deleted but data may still be within the system - future DeletionTime prop
3. all traces of the document permanently purged.

Reply:
I think I missed the fact that if `deletionTime` exists, it means the document is "logically" deleted already, hence the check. Thanks for the clarification!